fix potential Netty IO blocking in TripleProtocol connection handling…

ankitshokeen opened the pull request:
… #13041

## What is the purpose of the change?
This PR fixes Netty IO blocking in TripleProtocol's NettyConnectionClient to prevent client-side request timeouts and improve stability when using the triple protocol.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [x] Make sure gitHub actions can pass. [Why the workflow is failing and how to fix it?](../CONTRIBUTING.md)

ankitshokeen committed:
fix potential Netty IO blocking in TripleProtocol connection handling #13041

ankitshokeen committed:
Merge branch '3.3' into feature/netty-io-block-13041

ankitshokeen committed:
resolve slf4j binding errors in CI workflows

ankitshokeen committed:
upgrade java version 8 to 17

ankitshokeen committed:
Updated workflow

zrlw committed:
Log warning message at method exceptionCaught of Netty handlers (#15272)

dependabot[bot] committed:
Bump byte-buddy_version from 1.17.4 to 1.17.5 (#15285)

Bumps `byte-buddy_version` from 1.17.4 to 1.17.5.

Updates `net.bytebuddy:byte-buddy` from 1.17.4 to 1.17.5
- [Release notes](https://github.com/raphw/byte-buddy/releases)
- [Changelog](https://github.com/raphw/byte-buddy/blob/master/release-notes.md)
- [Commits](https://github.com/raphw/byte-buddy/compare/byte-buddy-1.17.4...byte-buddy-1.17.5)

Updates `net.bytebuddy:byte-buddy-agent` from 1.17.4 to 1.17.5
- [Release notes](https://github.com/raphw/byte-buddy/releases)
- [Changelog](https://github.com/raphw/byte-buddy/blob/master/release-notes.md)
- [Commits](https://github.com/raphw/byte-buddy/compare/byte-buddy-1.17.4...byte-buddy-1.17.5)

---
updated-dependencies:
- dependency-name: net.bytebuddy:byte-buddy
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: net.bytebuddy:byte-buddy-agent
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump io.netty.incubator:netty-incubator-codec-http3 (#15284)

Bumps [io.netty.incubator:netty-incubator-codec-http3](https://github.com/netty/netty-incubator-codec-http3) from 0.0.28.Final to 0.0.29.Final.
- [Commits](https://github.com/netty/netty-incubator-codec-http3/compare/netty-incubator-codec-http3-0.0.28.Final...netty-incubator-codec-http3-0.0.29.Final)

---
updated-dependencies:
- dependency-name: io.netty.incubator:netty-incubator-codec-http3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-surefire-plugin from 3.5.2 to 3.5.3 (#15286)

Bumps [org.apache.maven.plugins:maven-surefire-plugin](https://github.com/apache/maven-surefire) from 3.5.2 to 3.5.3.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.5.2...surefire-3.5.3)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

itaiit committed:
fix initialization problem when generateing wrapper object (#15276) (#15288)

dependabot[bot] committed:
Bump org.webjars:swagger-ui from 5.20.1 to 5.20.6 (#15300)

Bumps [org.webjars:swagger-ui](https://github.com/swagger-api/swagger-ui) from 5.20.1 to 5.20.6.
- [Release notes](https://github.com/swagger-api/swagger-ui/releases)
- [Changelog](https://github.com/swagger-api/swagger-ui/blob/master/.releaserc)
- [Commits](https://github.com/swagger-api/swagger-ui/compare/v5.20.1...v5.20.6)

---
updated-dependencies:
- dependency-name: org.webjars:swagger-ui
  dependency-version: 5.20.6
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jacoco:jacoco-maven-plugin from 0.8.12 to 0.8.13 (#15299)

Bumps [org.jacoco:jacoco-maven-plugin](https://github.com/jacoco/jacoco) from 0.8.12 to 0.8.13.
- [Release notes](https://github.com/jacoco/jacoco/releases)
- [Commits](https://github.com/jacoco/jacoco/compare/v0.8.12...v0.8.13)

---
updated-dependencies:
- dependency-name: org.jacoco:jacoco-maven-plugin
  dependency-version: 0.8.13
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump io.opentelemetry:opentelemetry-bom from 1.48.0 to 1.49.0 (#15298)

Bumps [io.opentelemetry:opentelemetry-bom](https://github.com/open-telemetry/opentelemetry-java) from 1.48.0 to 1.49.0.
- [Release notes](https://github.com/open-telemetry/opentelemetry-java/releases)
- [Changelog](https://github.com/open-telemetry/opentelemetry-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/open-telemetry/opentelemetry-java/compare/v1.48.0...v1.49.0)

---
updated-dependencies:
- dependency-name: io.opentelemetry:opentelemetry-bom
  dependency-version: 1.49.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump netty4_version from 4.1.119.Final to 4.2.0.Final (#15297)

Bumps `netty4_version` from 4.1.119.Final to 4.2.0.Final.

Updates `io.netty:netty-all` from 4.1.119.Final to 4.2.0.Final
- [Commits](https://github.com/netty/netty/compare/netty-4.1.119.Final...netty-4.2.0.Final)

Updates `io.netty:netty-bom` from 4.1.119.Final to 4.2.0.Final
- [Commits](https://github.com/netty/netty/compare/netty-4.1.119.Final...netty-4.2.0.Final)

---
updated-dependencies:
- dependency-name: io.netty:netty-all
  dependency-version: 4.2.0.Final
  dependency-type: direct:development
  update-type: version-update:semver-minor
- dependency-name: io.netty:netty-bom
  dependency-version: 4.2.0.Final
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

PengJingzhao committed:
fix(MetricsEventBus): fix some bugs (#15296)

RainYuY committed:
remove dubbo plugin useless dependency(dubbo-test-check) (#15236)

ankitshokeen commented:
Hi @iJIAJIA, @AlbumenJ, @JunJieLiu51520 could you please take a look and let me know if there are any further changes I should make? One of the CI tests is currently failing because it depends on a separate PR.
issue worked on https://github.com/apache/dubbo/issues/13041

cnzakii committed:
perf: Remove duplicate checks (#15327)

RainYuY committed:
Optimize GitHub Action to Retrieve target/logs/ Files (#15264)

Co-authored-by: Albumen Kevin <jhq0812@gmail.com>

zrlw committed:
Fixed OutOfMemoryError during maven install  (#15325)

* Fixed maven build argLine parameter

* remove maven-build-cache-extension to avoid OOM during maven build

* Set utf8 encoding to log chinese characters normally with different jdk

ankitshokeen committed:
Fix duplicate service mappings in TripleHttp3ProtocolTest

ankitshokeen committed:
Fix duplicate service mappings in TripleHttp3ProtocolTest

ankitshokeen committed:
Fix duplicate service mappings in TripleHttp3ProtocolTest

ankitshokeen committed:
Fix: formatting issue

AlbumenJ commented on pom.xml:
```diff
@@ -128,8 +128,8 @@
     <!-- Build args -->
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>17</maven.compiler.source>
+    <maven.compiler.target>17</maven.compiler.target>
```
Should not change this

ankitshokeen committed:
Fix: Use logger instead of System.out and properly handle cleanup exceptions in TripleHttp3ProtocolTest

ankitshokeen committed:
Refactor: rely on ApplicationModel.destroy for test cleanup

ankitshokeen committed:
fix potential Netty IO blocking in TripleProtocol connection handling

ankitshokeen committed:
Mergt
branch 'feature/netty-io-block-13041' of https://github.com/ankitshokeen/dubbo into feature/netty-io-block-13041

ankitshokeen committed:
fix potential Netty IO blocking in TripleProtocol connection handling
